manifests/quota.pp

```diff
@@ -0,0 +1,64 @@
+# @summary Install login scripts to run user quota commands.
+#
+# Install login scripts to run user quota commands.
+#
+# @param command_paths
+#   Paths to commands/scripts to run to display user
+#   quota information. Leave empty to NOT run any
+#   quota commands at login.
+#
+# @param file_prefix
+#   Run-script files will be added for various shells in /etc/profile.d/
+#   to wrap the quota scripts. This parameter specifies the prefix for
+#   these run-scripts. Will be applied as ${file_prefix}_quota.[extension].
+#
+# @param skip_users
+#   When these users log in quota commands will NOT be run.
+#
+# @param timeout_interval
+#   Quota commands are wrapped in 'timeout' and will be aborted after
+#   this number of seconds. Can be useful to prevent long "hangs" at
+#   login if there are filesystem problems.
+#
+# @example
+#   include profile_user_environment::quota
+class profile_user_environment::quota (
+  Array   $command_paths,
+  String  $file_prefix,
+  Array   $skip_users,
+  Integer $timeout_interval,
+) {
+  # defaults for file resources
+  File {
+    group => 'root',
+    mode => '0644',
+    owner => 'root',
+  }
+
+  $bash_file="/etc/profile.d/${file_prefix}_quota.sh"
+  $csh_file="/etc/profile.d/${file_prefix}_quota.csh"
+
+  if $command_paths.empty {
+    # no quota scripts to run, remove run-scripts
+    ## Bash-like shells
+    file { $bash_file:
+      ensure => absent,
+    }
+    ## csh-like shells
+    file { $csh_file:
+      ensure => absent,
+    }
```

Comment:
Yeah, but it does seem unlikely that the filenames would match exactly including the prefix. The workaround right now would be overriding the file_prefix parameter so it didn't match. But it would be hard to diagnose the problem in the first place if we weren't expecting something to be installing these files.